provider/github/watcher_test.go

```diff
@@ -133,17 +137,21 @@ func (s *WatcherTestSuite) TestWatch_WithError() {
 
 	s.NoError(err)
 
+	var wg sync.WaitGroup
+
 	err = w.Watch(context.TODO(), func(e lookout.Event) error {
-		switch e.Type() {
-		case pb.ReviewEventType:
-			s.Equal("96de6d9d321aa94faa0f053c72e2684a44961e5b", e.ID().String())
-		case pb.PushEventType:
-			s.Equal("d1f57cc4e520766576c5f1d9e7655aeea5fbccfa", e.ID().String())
-		}
+		wg.Add(1)
```

Comment:
this code is super weird. It calls `wg.Add` & `wg.Done` without any concurrency involved. There is only synchronous `s.Equal` check between them. Could you please help me understand how it works.

Reply:
Yes, this is a left over of other scaffolding things I was testing...
I put them trying to fix a failing assertion after the test ends:
`panic: Fail in goroutine after TestWatcherTestSuite/TestWatch_WithError has completed`

If the pull event goroutine returns the foo error, the push event goroutine will still call this callback before checking for `ctx.Done`.

Ideally the `Add` would be done in `(w *Watcher) watchEvents` before the `doEventRequest`. But putting `Add` works, I guess because it must be quicker than the `s.Equal`, and forces the `TestWatch_WithError` to wait before exiting.

I'll remove it because it's not technically correct. And this is just there to avoid a panic when an assertion fails, but that's not a concern if the test if failing anyway.

Reply:
changes pushed in ce7184b